extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -275,6 +277,58 @@ export class AnalyticsRoot {
     });
   }
 
+  /**
+   * @param {string} selector DOM query selector.
+   * @return {!Promise<!Array<!Element>>} Element corresponding to the selector.
+   */
+  getElements_(selector) {
+    // Wait for document-ready to avoid false missed searches
+    return this.ampdoc.whenReady().then(() => {
+      const results = [];
+      const foundElements = this.getRoot().querySelectorAll(selector);
+
+      // DOM search can "look" outside the boundaries of the root, thus make
+      // sure the result is contained. Length is not supported in all browsers
+      if (foundElements && foundElements.length) {
```

Comment:
This is not necessary, `querySelector` is already guaranteed to contain everything that is returned.

What you're probably thinking of is that `querySelector('div img')` can match on `<div><root><img/></root></div>`, but this contains check isn't going to fix that. Use `scopedQuerySelector` from `src/dom.js`